src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -169,9 +167,7 @@ public static DetailAST appendHiddenCommentNodes(DetailAST root) {
             DetailAST toVisit = curNode.getFirstChild();
             while (curNode != null && toVisit == null) {
                 toVisit = curNode.getNextSibling();
-                if (toVisit == null) {
-                    curNode = curNode.getParent();
-                }
+                curNode = curNode.getParent();
```

Comment:
Originally I thought this couldn't be right, but it does work out.
if `toVisit` is `null`, `curNode` is always replaced with the parent and the loop goes again.
if `toVisit` isn't `null`, it doesn't matter if `curNode` is replaced, the loop is broken and `curNode` is replaced with `toVisit`.